src/entity.ts

```diff
@@ -383,49 +384,101 @@ export class Entity {
 	}
 }
 
+export type EntityPropertiesOfType<T> = {
```

Comment:
_info_: Add `EntityPropertiesOfType<T>` type, where `T` is the "schema" of the type (the properties it has and their TS types. Lists are defined as non-nullable `ObservableArray<T>` and non-lists are defined as `T | null`.

Reply:
```ts
type Example = {
  text: string;
  values: number[];
  ref: OtherEntity;
  children: ChildEntity[];
};

type ExampleEntityProperties = EntityPropertiesOfType<Example> = {
  text: string | null;
  values: ObservableArray<number>;
  ref: OtherEntity | null;
  children: ObservableArray<ChildEntity>;
};
```

Reply:
Also, `readonly` properties are excluded, since they would represent calculated properties that can't be set.